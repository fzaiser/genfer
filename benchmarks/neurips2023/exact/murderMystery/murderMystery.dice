let aliceDunnit  = flip 0.3 in
let withGun = if aliceDunnit then flip 0.03 else flip 0.8 in
let tmp = observe withGun in
aliceDunnit
