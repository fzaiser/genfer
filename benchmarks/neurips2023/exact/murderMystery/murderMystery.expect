Support is a subset of: {0, ..., 1}

Computing moments...
Total measure:             Z = 0.569
Expected value:            E = 0.015817223198594025
Standard deviation:        Ïƒ = 0.12476793918663513
Variance:                  V = 0.015567038648879884
Skewness (3rd std moment): S = 7.761333239256879
Kurtosis (4th std moment): K = 61.23829365079365

Computing probabilities up to 2...
Unnormalized: p(0)     = 0.5599999999999999
Normalized:   p(0) / Z = 0.984182776801406
Unnormalized: p(1)     = 0.009
Normalized:   p(1) / Z = 0.015817223198594025
Unnormalized: p(n)     <= 0.0 for all n >= 2
Normalized:   p(n) / Z <= 0.0 for all n >= 2
