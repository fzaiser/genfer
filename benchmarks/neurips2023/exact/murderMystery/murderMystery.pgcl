aliceDunnit := bernoulli(0.3);
if(aliceDunnit = 1) {
    withGun := bernoulli(0.03);
} else {
    withGun := bernoulli(0.8);
}
observe(withGun = 1);

?Pr[aliceDunnit];
