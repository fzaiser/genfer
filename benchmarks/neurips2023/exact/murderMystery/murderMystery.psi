// flags: --dp
def main(){
    aliceDunnit := flip(3/10);
    withGun := if aliceDunnit { flip(3/100) } else { flip(4/5) };
    observe(withGun == true);
    return aliceDunnit;
}
