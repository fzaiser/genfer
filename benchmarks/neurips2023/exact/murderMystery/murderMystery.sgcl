aliceDunnit ~ Bernoulli(0.3);
if aliceDunnit = 1 {
    withGun ~ Bernoulli(0.03);
} else {
    withGun ~ Bernoulli(0.8);
}
observe withGun = 1;

return aliceDunnit;
