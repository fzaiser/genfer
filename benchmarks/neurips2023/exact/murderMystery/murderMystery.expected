0.01581722
9/569
